src/main/java/kayu/storage/Storage.java

```diff
@@ -0,0 +1,158 @@
+    protected static final String UNABLE_TO_CREATE_FILE = "Load/save file ./%s cannot be created.";
+    protected static final String UNABLE_TO_LOAD_PATH = "Path ./%s cannot be accessed/loaded.";
+    protected static final String INVALID_TASK_FORMAT = "'%s' is an invalid Task entry.";
+    
+    // Default task file directory.
+    private String taskDirectoryPath = "data";
+    private String taskFilePath = taskDirectoryPath + "/task_list.txt";
+
+    /**
+     * Sets the desired file path of {@link kayu.task.Task} storage.
+     *
+     * @param taskFilePath File path for reading/writing of data.
+     */
+    public void setDirectoryAndFilePath(String taskFilePath) {
+        setTaskFilePath(taskFilePath);
+        
+        int splitIdx = taskFilePath.lastIndexOf('/'); // mac/unix
+        if (splitIdx < 0) {
+            splitIdx = taskFilePath.lastIndexOf('\\'); // win
+        }
+        setTaskDirectoryPath(taskFilePath.substring(0, splitIdx + 1));
+    }
+
+    protected void setTaskDirectoryPath(String taskDirectoryPath) {
+        this.taskDirectoryPath = taskDirectoryPath;
+    }
+
+    protected void setTaskFilePath(String taskFilePath) {
+        this.taskFilePath = taskFilePath;
+    }
+
+    /**
+     * Loads the file based on specified directory, extracts and decodes the
+     * {@link kayu.task.Task} instances for return.
+     *
+     * @return List of {@link kayu.task.Task}.
+     * @throws StorageException If unable to read/write to file.
+     */
+    public List<Task> load() throws StorageException {
+        initializeFilePath();
+        List<String> taskLines = readFile();
+        return decodeAll(taskLines);
+    }
+
+    protected void initializeFilePath() throws StorageException {
+        try {
+            File directory = new File(taskDirectoryPath);
+            if (!directory.exists() && !directory.mkdir()) {
+                throw new StorageException(String.format(UNABLE_TO_CREATE_DIRECTORY, taskDirectoryPath));
+            }
+            
+            File file = new File(taskFilePath);
+            if (!file.exists() && !file.createNewFile()) {
+                throw new StorageException(String.format(UNABLE_TO_CREATE_FILE, taskFilePath));
+            }
+        } catch (IOException exception) {
+            throw new StorageException(String.format(UNABLE_TO_LOAD_PATH, taskFilePath));
+        }
+    }
+
+    protected List<String> readFile() throws StorageException {
+        try {
+            Path filePath = Paths.get(taskFilePath);
+            return Files.readAllLines(filePath);
+            
+        } catch (IOException exception) {
+            throw new StorageException(String.format(UNABLE_TO_LOAD_PATH, taskFilePath));
+        }
+    }
+
+    protected List<Task> decodeAll(List<String> taskLines) throws StorageException {
+        List<Task> savedTaskList = new ArrayList<>();
+        
+        for (String stringTask : taskLines) {
```

Comment:
You might want to consider standardising the syntax for this kind of **for loop** statements. Most of the other segments of your code do not have a whitespace before the colon, you might want to consider standardising to that format 😄

Reply:
Yeap good catch! Removed whitespace.